domains/certificates/RegistryConnector/Worker.UnitTests/ProjectOriginRegistryEventHandlerTests.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using Contracts.Certificates;
+using FluentAssertions;
+using MassTransit;
+using MassTransit.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+using ProjectOrigin.Electricity.Client.Models;
+using RegistryConnector.Worker.Cache;
+using RegistryConnector.Worker.EventHandlers;
+using Xunit;
+
+namespace RegistryConnector.Worker.UnitTests;
+
+public class ProjectOriginRegistryEventHandlerTests
+{
+    private readonly PoRegistryEventHandlerFixture fixture;
+
+    public ProjectOriginRegistryEventHandlerTests()
+    {
+        fixture = new PoRegistryEventHandlerFixture();
+    }
+
+    [Fact]
+    public async void ShouldRetryCachePop10Times()
+    {
+        var target = fixture
+            .WithNotFoundInCache()
+            .BuildTarget();
+
+        await target.OnRegistryEvents(new CommandStatusEvent(Some.CommandId, CommandState.Succeeded, null));
+
+        fixture.Verify10CallsToPop();
+    }
+
+    [Fact]
+    public async void ShouldCallBusPublishOnSuccessfulCachePop()
+    {
+        await using var provider = new ServiceCollection()
+            .AddMassTransitTestHarness()
+            .BuildServiceProvider(true);
+        var harness = provider.GetRequiredService<ITestHarness>();
+        await harness.Start();
+
+        var wrappedEvent = new MessageWrapper<ProductionCertificateCreatedEvent>(Some.ProductionCertificateCreatedEvent, Guid.NewGuid(), Guid.NewGuid());
+        var target = fixture
+            .WithFoundInCache(wrappedEvent)
+            .BuildTarget(harness.Bus);
+
+        await target.OnRegistryEvents(new CommandStatusEvent(Some.CommandId, CommandState.Succeeded, null));
+
+        harness.Published.Count().Should().Be(1);
+        fixture.VerifyOneCallToCachePop();
+    }
+
+    private class PoRegistryEventHandlerFixture
+    {
+        private readonly Mock<ILogger<ProjectOriginRegistryEventHandler>> loggerMock;
+        private readonly Mock<ICertificateEventsInMemoryCache> cacheMock;
+        private readonly Mock<IBus> busMock;
```

Comment:
I would prefer that the MassTransit harness is only used in the fixture

Reply:
is now in fixture